### PR TITLE
220166_Align-Invoice-API-Models-PaymentRequestBatch

### PR DESCRIPTION
**Pull request by fshehzad-defra**
220166 align invoice api models payment request batch

### CODE REVIEW

File: EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs [hunk 1]
```diff
@@ -19,7 +19,7 @@ public class InvoiceLineValidatorTests
         private readonly FieldsRoute _route = new()
         {
             PaymentType = "AP",
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/InvoiceLineValidatorTests.cs:21]: PaymentType should be the last of the four. (It's for 'EU' or 'DOM')

File: EST.MIT.Invoice.Api.Test/Repositories/ReferenceDataRepository_GetFundCodeListAsync.cs [hunk 2]
```diff
@@ -53,7 +53,7 @@ public void GetFundCodesListAsync_Returns_200()
         [InlineData("", "EST", "AP", "BPS")]
         [InlineData("RPA", "EST", "", "BPS")]
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/Repositories/ReferenceDataRepository_GetFundCodeListAsync.cs:54]: Would really like to see more accurate test data - see the routes table.

File: EST.MIT.Invoice.Api.Test/Repositories/ReferenceDataRepository_GetPaymentTypesListAsync_Tests.cs [hunk 3]
```diff
@@ -9,7 +9,7 @@ namespace EST.MIT.Invoice.Api.Test.Repositories;
 public class ReferenceDataRepository_GetPaymentTypesListAsync_Tests
 {
     private readonly Mock<HttpMessageHandler> _mockHttpMessageHandler;
-    private string _invoiceType = "RPA";
+    private string _accountType = "RPA";
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api.Test/Repositories/ReferenceDataRepository_GetPaymentTypesListAsync_Tests.cs:12]: See my other 'more accurate test data values' comment

**Review by gldefra — APPROVED** [review]
LGTM

File: EST.MIT.Invoice.Api/Repositories/Interfaces/IReferenceDataRepository.cs [hunk 4]
```diff
@@ -2,10 +2,10 @@ namespace EST.MIT.Invoice.Api.Repositories.Interfaces;
...
-    Task<HttpResponseMessage> GetSchemeTypesListAsync(string? invoiceType, string? organisation);
-    Task<HttpResponseMessage> GetPaymentTypesListAsync(string? invoiceType, string? organisation, string? schemeType);
-    Task<HttpResponseMessage> GetOrganisationsListAsync(string? invoiceType);
-    Task<HttpResponseMessage> GetSchemeCodesListAsync(string? invoiceType, string? organisation, string? paymentType, string? schemeType);
-    Task<HttpResponseMessage> GetFundCodesListAsync(string? invoiceType, string? organisation, string? paymentType, string? schemeType);
-    Task<HttpResponseMessage> GetCombinationsListForRouteAsync(string invoiceType, string organisation, string paymentType, string schemeType);
+    Task<HttpResponseMessage> GetSchemeTypesListAsync(string? accountType, string? organisation);
+    Task<HttpResponseMessage> GetPaymentTypesListAsync(string? accountType, string? organisation, string? schemeType);
+    Task<HttpResponseMessage> GetOrganisationsListAsync(string? accountType);
+    Task<HttpResponseMessage> GetSchemeCodesListAsync(string? accountType, string? organisation, string? paymentType, string? schemeType);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Repositories/Interfaces/IReferenceDataRepository.cs:8]: I'd prefer if paymentType and schemeType were the other way round, so paymentType is last.

File: EST.MIT.Invoice.Api/Services/Api/Interfaces/IReferenceDataApi.cs [hunk 5]
```diff
@@ -4,9 +4,9 @@
...
 {
-    Task<ApiResponse<IEnumerable<PaymentScheme>>> GetSchemeTypesAsync(string? invoiceType, string? organisation);
-    Task<ApiResponse<IEnumerable<PaymentType>>> GetPaymentTypesAsync(string? invoiceType, string? organisation, string? schemeType);
-    Task<ApiResponse<IEnumerable<Organisation>>> GetOrganisationsAsync(string? invoiceType);
-    Task<ApiResponse<IEnumerable<SchemeCode>>> GetSchemeCodesAsync(string? invoiceType, string? organisation, string? paymentType, string? schemeType);
-    Task<ApiResponse<IEnumerable<FundCode>>> GetFundCodesAsync(string? invoiceType, string? organisation, string? paymentType, string? schemeType);
+    Task<ApiResponse<IEnumerable<PaymentScheme>>> GetSchemeTypesAsync(string? accountType, string? organisation);
+    Task<ApiResponse<IEnumerable<PaymentType>>> GetPaymentTypesAsync(string? accountType, string? organisation, string? schemeType);
+    Task<ApiResponse<IEnumerable<Organisation>>> GetOrganisationsAsync(string? accountType);
+    Task<ApiResponse<IEnumerable<SchemeCode>>> GetSchemeCodesAsync(string? accountType, string? organisation, string? paymentType, string? schemeType);
```
Comment (andygibbonsdefra) [inline EST.MIT.Invoice.Api/Services/Api/Interfaces/IReferenceDataApi.cs:10]: Would prefer paymentType and schemeType the other way around.

**Pull request merged**